modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -544,148 +564,183 @@ sub print_diagram {
 sub apply_tweaks {
     my $self    = shift @_;
     my $tweaks  = shift @_;
-
+    my @response;
+    my $responseStructure;
     my $need_write = 0;
-
+    $responseStructure->{Tweaks} = [];
     foreach my $tweak (@$tweaks) {
-        print "\nTweak.Request\t$tweak\n";
+        push @response, "\nTweak.Request\t$tweak\n";
 
         if($tweak=~/^pipeline\.param\[(\w+)\](\?|#|=(.+))$/) {
             my ($param_name, $operator, $new_value_str) = ($1, $2, $3);
-
             my $pwp_collection  = $self->collection_of( 'PipelineWideParameters' );
             my $hash_pair       = $pwp_collection->find_one_by('param_name', $param_name);
+            my $value           = $hash_pair ? $hash_pair->{'param_value'} : undef;
+            my $tweakStructure;
+            $tweakStructure->{Action} = $self->{TWEAK_ACTION}->{substr($operator, 0, 1)};
+            $tweakStructure->{Object}->{Type} =  $self->{TWEAK_OBJECT_TYPE}->{PIPELINE};
+            $tweakStructure->{Object}->{Id} = undef;
+            $tweakStructure->{Object}->{Name} = undef;
+            $tweakStructure->{Return}->{Field} = $param_name;
+            $tweakStructure->{Return}->{OldValue} = $value;
 
             if($operator eq '?') {
-                print "Tweak.Show    \tpipeline.param[$param_name] ::\t"
-                     . ($hash_pair ? $hash_pair->{'param_value'} : '(missing_value)') . "\n";
+                $tweakStructure->{Return}->{NewValue} = $value;
+                push @response, "Tweak.Show    \tpipeline.param[$param_name] ::\t"
+                    . ($hash_pair ? $hash_pair->{'param_value'} : '(missing_value)') . "\n";
```

Comment:
Don't think I'm enjoying it, but this is still not right

Reply:
Before my changes intentions were different here and on line 669 (5 here and 4 there), rather confusing. On line 669 you commented to do is it was before. Should I do 4 or 5 spaces???